packages/ui/src/app/pages/Election/components/ElectionProgressBar.tsx

```diff
@@ -0,0 +1,459 @@
+import {
+  MultiStatisticItem,
+  StatisticItemSpacedContent,
+  StatisticLabel,
+  StatisticItemProps,
+} from '@/common/components/statistics'
+import { TooltipText } from '@/common/components/Tooltip'
+import { DurationValue } from '@/common/components/typography/DurationValue'
+import { AN_HOUR, A_DAY, A_MINUTE, BorderRad, Colors, Transitions, ZIndex } from '@/common/constants'
+import { splitDuration, MILLISECONDS_PER_BLOCK } from '@/common/model/formatters'
+import { toNumber } from '@/common/utils'
+import { useCouncilConstants } from '@/council/hooks/useCouncilConstants'
+
+interface ElectionProgressBarProps extends StatisticItemProps {
+  electionStage: string
+  value?: number | BN
+  currentBlock?: number | BN
+}
+
+export const ElectionProgressBar = (props: ElectionProgressBarProps) => {
+  const currentBlock = toNumber(props.currentBlock)
+  const duration = toNumber(props.value)
+
+  const convertDurationToTimeString = (duration: number) => {
+    const format = splitDuration([
+      [A_DAY, 'd'],
+      [AN_HOUR, 'h'],
+      [A_MINUTE, 'm'],
+    ])
+
+    const formatDurationDate = (duration: number): [string | number, string][] => {
+      if (duration * MILLISECONDS_PER_BLOCK < A_MINUTE) {
+        return [['< 1', 'm']]
+      }
+      return format(duration * MILLISECONDS_PER_BLOCK)
+    }
+
+    return <DurationValue value={formatDurationDate(duration)} />
+  }
+
+  const endDayOfStage = useMemo(() => convertDurationToTimeString(duration), [duration])
+
+  const [stageDescription, setStageDescription] = useState(props.electionStage)
+  const [verbIndicator, setVerbIndicator] = useState('ends in')
+  const [daysIndicator, setDaysIndicator] = useState<any>(endDayOfStage)
+  const [selectedToolbarStage, setSelectedToolbarStage] = useState('')
+
+  useEffect(() => {
+    if (selectedToolbarStage === '') updateDescription(selectedToolbarStage, false)
+    else updateDescription(selectedToolbarStage, true)
+  }, [endDayOfStage])
+
+  let announcingProgress = 0
+  let votingProgress = 0
+  let revealingProgress = 0
+  let inactiveProgress = 0
+
+  let remainDays = 0
+  let announcingDays = 0
+  let votingDays = 0
+  let revealingDays = 0
+  let inactiveDays = 0
+
+  let announcingEndDay = ''
+  let votingEndDay = ''
+  let revealingEndDay = ''
+  let inactiveEndDay = ''
+
+  let announcingEndBlock = 0
+  let votingEndBlock = 0
+  let revealingEndBlock = 0
+  let inactiveEndBlock = 0
+
+  let progressBarAttr = '1fr 14fr 3fr 3fr'
+
+  const constants = useCouncilConstants()
+
+  const blockDurationToMs = (blockDuration: number) => blockDuration * MILLISECONDS_PER_BLOCK
+  const blockToDate = (duration: number) => {
+    const now = Date.now()
+    const msDuration = blockDurationToMs(duration)
+    return new Date(now + msDuration).toLocaleString('en-gb', { timeZone: 'Europe/Paris' })
+  }
+  const blockDurationToDays = (blockDuration: number) => Math.floor(blockDurationToMs(blockDuration) / A_DAY)
```

Comment:
Please move this functions out of the component context (`blockDurationToMs`, `blockToDate`, `blockDurationToDays`). They can just be in the module context directly since they don't rely on some values from the component context.